habitat_baselines/rl/ddppo/algo/ddp_utils.py

```diff
@@ -29,6 +31,46 @@
 )
 
 
+def is_slurm_job() -> bool:
+    return SLURM_JOBID is not None
+
+
+def is_slurm_batch_job() -> bool:
+    return is_slurm_job() and os.environ.get("SLURM_JOB_NAME", "bash") not in (
+        "bash",
+        "zsh",
+        "fish",
+        "tcsh",
+        "sh",
+    )
+
+
+@overload
+def rank0_only() -> bool:
+    ...
+
+
+@overload
+def rank0_only(fn: Callable) -> Callable:
+    ...
+
+
+def rank0_only(fn: Optional[Callable] = None) -> Union[Callable, bool]:
```

Comment:
Really like the elegance that wrapper bring to our code base. Let's add the docstring that this decorator designed to wrap methods that are allowed called only on master worker if distrib setting is used. Thank you!